### PR TITLE
Chore/update drpc message formats part 2

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [ ] Bug Fix
- [x] Chore
- [ ] Feature
- [ ] Documentation Update
- [ ] Code style update (formatting, local variables)
- [ ] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets

VR-235

## High level description

Fixes up the drpc request/response formats in line with https://github.com/digicatapult/veritable-documentation/tree/main/schemas/veritable_messaging/0.1. Note this isn't perfect due to some issue with that documentation that will be updated in a PR there as part of VR-298

## Detailed description

In the main this adds the `createdTime` and `expiresTime` fields which are in seconds since unix epoch. However this does include several other small fixes:

* type checking was disabled for the tests, this is now enabled along with a bunch of small fixes
* I've moved the `veritableCloudagentInt` into the `veritableCloudagent` folder so it's better contained. This then updates a bunch of imports

I have separated these into their own commits so feel free to review these commit by commit

Some of the discrepencies wrt the documentation are:

* subjectId is now a discriminated type union to support more subjectId formats
* response formats use `result` and `error` properties as per jsonrpc

Some example messages:

```ts
submitDrpcRequest = {
  "jsonrpc": "2.0",
  "method": "submit_query_request",
  "params": {
    "id": "03f3e4e7-47b6-4bf6-9076-ad4cc361b52c",
    "createdTime": 1730994619,
    "expiresTime": 1731599419,
    "type": "https://github.com/digicatapult/veritable-documentation/tree/main/schemas/veritable_messaging/query_types/total_carbon_embodiment/request/0.1",
    "data": {
      "subjectId": {
        "idType": "product_and_quantity",
        "content": {
          "productId": "Test1",
          "quantity": 1
        }
      }
    }
  }
}

submitDrpcRequest = {
  "jsonrpc": "2.0",
  "method": "submit_query_response",
  "params": {
    "id": "0e55b8fb-436e-4c7e-aec2-0f3e30b99dd5",
    "createdTime": 1730994514,
    "expiresTime": 1731599314,
    "type": "https://github.com/digicatapult/veritable-documentation/tree/main/schemas/veritable_messaging/query_types/total_carbon_embodiment/response/0.1",
    "data": {
      "subjectId": {
        "idType": "product_and_quantity",
        "content": {
          "productId": "Test2",
          "quantity": 2
        }
      },
      "mass": 1,
      "unit": "kg",
      "partialResponses": []
    }
  }
}

submitDrpcResponse = {
  "jsonrpc": "2.0",
  "result": {
    "type": "https://github.com/digicatapult/veritable-documentation/tree/main/schemas/veritable_messaging/query_ack/0.1"
  }
}

```

## Describe alternatives you've considered

None

## Operational impact

This will nuke all queries in the demo environment. This should not be a problem

## Additional context

None

### CODE REVIEW

File: src/controllers/queries/__tests__/index.test.ts [hunk 1]
```diff
@@ -3,14 +3,23 @@ import { describe, it } from 'mocha'
...
 describe('QueriesController', () => {
   const req = { log: mockLogger } as unknown as Request
+  let clock: sinon.SinonFakeTimers | null = null
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers({ now: 1000, toFake: ['Date'] })
+  })
 
   afterEach(() => {
+    clock?.restore()
```
Comment (Ellenn-A) [inline src/controllers/queries/__tests__/index.test.ts:22]: if clock is always created in before each, why are we checking if it even exists before restoring? Won't it always exist?

Comment (mattdean-digicatapult) [reply src/controllers/queries/__tests__/index.test.ts:22]: we know that but the type system doesn't. Hence the `?` to convince it that it's all fine

Comment (Ellenn-A) [reply src/controllers/queries/__tests__/index.test.ts:22]: ok

**Review by Ellenn-A — COMMENTED** [review]
I didn't spot anything but maybe it'd be good if @n3op2 also has a look before the merge?

File: src/controllers/queries/__tests__/index.test.ts [hunk 2]
```diff
@@ -526,8 +575,15 @@ describe('QueriesController', () => {
...
+              unit: 'kg',
               partialResponses: [],
-              subjectId: '00000000-0000-0000-0000-d8ae0805059e',
+              subjectId: {
+                idType: 'product_and_quantity',
+                content: {
+                  productId: '00000000-0000-0000-0000-d8ae0805059e',
+                  quantity: 2,
+                },
+              },
```
Comment (doowt) [inline src/controllers/queries/__tests__/index.test.ts:586]: Need to be clear in documentation when it's updated that the response value (here 10kg) means the total for the subject (in this case it is the cumulative total for two objects (quantity 2)).

**Review by doowt — DISMISSED** [review]

File: src/controllers/queries/__tests__/index.test.ts (hunk 2 above)
Comment (doowt) [reply src/controllers/queries/__tests__/index.test.ts:586]: Added comment to VR-297

File: src/models/drpc.ts [hunk 3]
```diff
@@ -2,12 +2,24 @@ import z from 'zod'
 
 import { UUID } from './strings.js'
 
+const subjectIdParser = z.discriminatedUnion('idType', [
+  z.object({
+    idType: z.literal('product_and_quantity'),
```
Comment (n3op2) [inline src/models/drpc.ts:7]: I guess with each query we will add another one with it's own specific content. E.g. provenance will contain it's own?

File: src/models/drpc.ts [hunk 4]
```diff
@@ -2,12 +2,24 @@ import z from 'zod'
...
+
 export const carbonEmbodimentRequestData = z.object({
-  subjectId: z.string(),
-  quantity: z.number().int().min(1),
+  subjectId: subjectIdParser,
 })
 export const carbonEmbodimentRequest = z.object({
   id: z.string(),
+  createdTime: z.number().int().gte(0),
+  expiresTime: z.number().int().gte(0),
```
Comment (n3op2) [inline src/models/drpc.ts:22]: I assume this will be in unix format but why still not date?

File: src/views/queries/responseCo2scope3.tsx [hunk 5]
```diff
@@ -172,9 +172,24 @@ export default class Scope3CarbonConsumptionResponseTemplates {
...
+      case 'mg':
+        return 1e-6
+      case 'g':
+        return 1e-3
+      case 'kg':
+        return 1
+      case 'tonne':
+        return 1e3
+    }
+  }
```
Comment (n3op2) [inline src/views/queries/responseCo2scope3.tsx:188]: wonder how the exponential notation will work if it exceeds a number size e.g. how we are going to render to the user.

File: src/views/queries/responseCo2scope3.tsx [hunk 6]
```diff
@@ -172,9 +172,24 @@ export default class Scope3CarbonConsumptionResponseTemplates {
...
+
   private reduceResponse = (query: CarbonEmbodimentRes['data']): number => {
     return (
-      query.mass +
+      query.mass * this.multFactor(query.unit) +
```
Comment (n3op2) [inline src/views/queries/responseCo2scope3.tsx:192]: where do we use query.mass? must have missed that one

File: src/models/drpc.ts (hunk 3 above)
Comment (mattdean-digicatapult) [reply src/models/drpc.ts:7]: correct

**Review by n3op2 — APPROVED** [review]
Left a few questions in regards to the new properties but nothing major that would stop me from approving

File: src/views/queries/responseCo2scope3.tsx (hunk 5 above)
Comment (mattdean-digicatapult) [reply src/views/queries/responseCo2scope3.tsx:188]: this is just syntax. Javascript doesn't have the concept of decimal

File: src/views/queries/responseCo2scope3.tsx (hunk 6 above)
Comment (mattdean-digicatapult) [reply src/views/queries/responseCo2scope3.tsx:192]: Quite a lot of places. This replaces `emissions` elsewhere

**Pull request merged**